airflow/api_fastapi/core_api/routes/public/dag_report.py

```diff
@@ -41,16 +46,30 @@
             status.HTTP_400_BAD_REQUEST,
         ]
     ),
+    dependencies=[Depends(requires_access_dag(method="GET"))],
 )
 def get_dag_reports(
     subdir: str,
+    readable_dags_filter: ReadableDagsFilterDep,
 ):
     """Get DAG report."""
     fullpath = os.path.normpath(subdir)
     if not fullpath.startswith(settings.DAGS_FOLDER):
         raise HTTPException(status.HTTP_400_BAD_REQUEST, "subdir should be subpath of DAGS_FOLDER settings")
+
     dagbag = DagBag(fullpath)
+
+    readable_dag_ids: set[str] | None = readable_dags_filter.value
+    if readable_dag_ids:
+        filtered_dagbag_stats = [
+            file_load_stat
+            for file_load_stat in dagbag.dagbag_stats
+            if len(set(ast.literal_eval(file_load_stat.dags)) - readable_dag_ids) == 0
```

Comment:
Not 100% sure whether it's a good idea to use `ast.literal_eval`. But we use it elsewhere in airflow and I think it should be better than `eval`

Reply:
+1